packages/rest/src/body-parsers/body-parser.json.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {RequestBodyParserOptions} from '../types';
+
+import {json} from 'body-parser';
+import {inject} from '@loopback/context';
+import {Request} from '../..';
+import {RestBindings} from '../keys';
+import {is} from 'type-is';
+
+import {RequestBody, BodyParser} from './types';
+import {
+  BodyParserWithCallback,
+  DEFAULT_LIMIT,
+  getParserOptions,
+  parseRequest,
+} from './helper';
+
+export class JsonBodyParser implements BodyParser {
+  name = 'json';
+  private jsonParser: BodyParserWithCallback;
+
+  constructor(
+    @inject(RestBindings.REQUEST_BODY_PARSER_OPTIONS, {optional: true})
+    options: RequestBodyParserOptions = {},
+  ) {
+    const jsonOptions = Object.assign(
```

Comment:
Can you move this part into `getParserOptions` too, to simplify our parsers into something like this:

```ts
const jsonOptions = getParserOptions(
  'json', // the key (required)
  {type: 'json', limit: DEFAULT_LIMIT}, // the defaults (required)
  options // user-provided options (optional)
);
```

Do we actually have to explicitly set `{type: 'json'}`? Can `getParserOptions` set that property for us? And how about `limit`, isn't this something that `getParserOptions` can provide default value for too?

```ts
const jsonOptions = getParserOptions(
  'json', // the key (required)
  {limit: DEFAULT_LIMIT}, // the defaults (required)
  options // user-provided options (optional)
);